src/plugins/MamicoCoupling.h

```diff
@@ -2,114 +2,135 @@
+ * simulation from both the outside and inside. From the outside, MaMiCo needs
+ * to be able to start and stop the simulation, and keep unique simulations
+ * distinct. From the inside, MaMiCo code needs to be executed at specific
+ * points within the same simulation step. This plugin enables the "inside"
+ * behaviour.
+ *
+ * MaMiCo coupling requires the MAMICO_COUPLING flag to be set, and the
+ * MAMICO_SRD_DIR flag to point to MaMiCo source files. With these enabled, ls1
+ * compiles as a library. To make sure the program compiles and works with
+ * tests, the relevant MaMiCo portions for the code are put in #ifdef regions.
+ */
+
+class MamicoCoupling : public PluginBase {
 
 public:
-	MamicoCoupling() {}
-	virtual ~MamicoCoupling() {}
-
-	/** @brief Prints to log that mamico coupling is initialized
-	 * 
-	 * */
-	void init(ParticleContainer* particleContainer,
-			DomainDecompBase* domainDecomp, Domain* domain) override;
-
-
-	void readXML(XMLfileUnits& xmlconfig) override;
-
-	void beforeEventNewTimestep(
-			ParticleContainer* particleContainer, DomainDecompBase* domainDecomp,
-			unsigned long simstep
-	) override;
-
-	/** @brief Takes coupling steps such as particle insertion, to make sure they are accounted for before forces are calculated.
-	 * 
-	 * Following steps are taken, if coupling is switched on:
-	 * - Iterate over cells to average values like momentum and mass, to pass to macroscopic solvers
-	 * - Distribute incoming mass from macroscopic solver by inserting perticles (if enabled)
-	 * - Run the MaMiCo thermostat cell by cell
-	 * 
-	 * The distributeMass method calls the updateParticleContainerAndDecomposition() function at the end, so we end up with a container with halo particles present.
-	 * Hence a manual halo clearance is done to restore the state of the container.
-	 * */
-	void beforeForces(
-			ParticleContainer* particleContainer, DomainDecompBase* domainDecomp,
-			unsigned long simstep
-	) override;
-
-	/** @brief Performs adjustments after force calculation 
-	 * 
-	 * Following steps are taken, if coupling is switched on:
-	 * - Distribute incoming momentum among affected cells
-	 * - Apply boundary force to molecules near microscopic domain boundary
-	 * */	
-	void afterForces(
-			ParticleContainer* particleContainer, DomainDecompBase* domainDecomp,
-			unsigned long simstep
-	) override;
-
-	void endStep(
-			ParticleContainer* particleContainer, DomainDecompBase* domainDecomp,
-			Domain* domain, unsigned long simstep) override;
-
-	void finish(ParticleContainer* particleContainer,
-			DomainDecompBase* domainDecomp, Domain* domain) override;
-
-	std::string getPluginName() override {
-		return std::string("MamicoCoupling");
-	}
-
-	static PluginBase* createInstance() { return new MamicoCoupling(); }
+  MamicoCoupling() {}
+  virtual ~MamicoCoupling() {}
+
+  /**
+   * Prints to log that mamico coupling is initialized
+   */
```

Comment:
Comment superfluous or even misleading if init does something useful besides printing log messages ...

Reply:
I've added "currently only", should I keep it like that or remove the comment altogether?

Reply:
Just remove the comment entirely.